Formula/m/micro.rb

```diff
@@ -20,7 +20,8 @@ class Micro < Formula
   depends_on "go" => :build
 
   def install
-    system "make", "build-tags"
+    system "git", "fetch", "--tags", "--force"
```

Comment:
What does it need all other tags for when it has the tag checkout already?

Reply:
When `micro` is build for something different than a release (e.g. `--HEAD`, see zyedidia/micro#3447) then it creates a relative version number of the last tagged release + followup commits. Therefore it needs all tags present.
The current problem isn't the absence of a tag, but one "moving" tag (`nightly`) instead, which must be overwritten.

Reply:
So this is to make `--HEAD` builds include proper version information?

Reply:
Correct, but it's just half of the truth, since the tags are necessary for the `release` builds too (it's not hard-coded).
Since the formula invokes `make build-tags` it implicitly fetches all tags from `micro`'s build process and I assume, that it fails with `release` builds too in that moment, when a "old" nightly tag is present.

For that reason we did half of the way [upstream](https://github.com/zyedidia/micro/pull/3448), but need to fix the formula too to bypass trouble with e.g. `v2.0.14` resp. already fetched `nightly` tag.

Reply:
So how to make it not fetch anything and just build with what's in the checkout?

Reply:
`make build` does exactly that.
But without any tags [build-version.go#L41](https://github.com/zyedidia/micro/blob/master/tools/build-version.go#L41) and below isn't able to properly name itself.

Reply:
Any further thoughts?
Otherwise it should work as is with `v2.0.15` again.

Reply:
Does the problem go away if you do
```
rm -rf "$(brew --cache)/micro--git"
```
before updating your `--HEAD` install?

If so then this seems like something that should be fixed in `brew` instead of in the formula.

Reply:
If that means to remove the whole `git` history from that formula source (what I assume), then it should work too.
The problem is simply caused by an already present `tag` (`nightly`) in the already checked out formula source folder, which changes over time (daily).